be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -674,129 +716,389 @@ Status JsonReader::_read_and_parse_json() {
+            }
+            _construct_numeric_column(column, i16);
+            return Status::OK();
         }
 
-        case simdjson::ondemand::number_type::floating_point_number: {
-            std::string_view sv;
-            err = simdjson::to_json_string(value).get(sv);
-            if (UNLIKELY(err)) {
-                column->append_nulls(1);
-                break;
+        case TYPE_TINYINT: {
+            int8_t i8 = 0;
+            if (cast_int64(i64, &i8) != 0) {
+                std::string err_msg = strings::Substitute(
+                        "Overflow in cast the integer value as TINYINT. value=$0, column=$1", i64, desc->col_name());
+                return Status::DataQualityError(err_msg.c_str());
             }
+            _construct_numeric_column(column, i8);
+            return Status::OK();
+        }
 
-            column->append_strings(std::vector<Slice>{Slice{sv.data(), sv.size()}});
-            break;
+        // Float column, integer value.
+        case TYPE_FLOAT: {
+            auto f = static_cast<float>(i64);
+            _construct_numeric_column(column, f);
+            return Status::OK();
+        }
+
+        // Double column, integer value.
+        case TYPE_DOUBLE: {
+            auto d = static_cast<double>(i64);
+            _construct_numeric_column(column, d);
+            return Status::OK();
         }
 
         default: {
-            column->append_nulls(1);
-            break;
+            std::string err_msg =
+                    strings::Substitute("Unable to cast the integer value. value=$0, column=$1", i64, desc->col_name());
+            return Status::DataQualityError(err_msg.c_str());
         }
         }
-        break;
-    }
 
-    case simdjson::ondemand::json_type::string: {
-        std::string_view sv;
-        auto err = value.get_string().get(sv);
+    } else if (tp == simdjson::ondemand::number_type::floating_point_number) {
+        double d = 0;
+        err = value.get_double().get(d);
         if (UNLIKELY(err)) {
-            column->append_nulls(1);
-            break;
+            std::string err_msg =
+                    strings::Substitute("Failed to parse the numeric column value as double. code=$0, error=$1", err,
+                                        simdjson::error_message(err));
+            return Status::DataQualityError(err_msg.c_str());
         }
 
-        column->append_strings(std::vector<Slice>{Slice(sv.data(), sv.length())});
-        break;
+        switch (type_desc.type) {
+        // Double column, float value.
+        case TYPE_DOUBLE: {
+            _construct_numeric_column(column, d);
+            return Status::OK();
+        }
+
+        // Float column, float value.
+        case TYPE_FLOAT: {
+            auto f = static_cast<float>(d);
+            _construct_numeric_column(column, f);
+            return Status::OK();
+        }
+
+        case TYPE_BIGINT: {
+            auto i64 = static_cast<int64_t>(d);
```

Comment:
Need to check overflow or underflow, adding a unit test to cover overflow and underflow would be better